### ISSUE TITLE
GithubBrowserSample ViewDataBinding Leak?

### ISSUE THREAD

**Issue by timatoe** [android/architecture-components-samples#673]
I've added Leak Canary to GithubBrowserSample and it has detected the following leak: 

```
2019-07-10 14:22:07.507 11220-12066/com.android.example.github D/LeakCanary: HeapAnalysisSuccess(heapDumpFile=/data/user/0/com.android.example.github/files/leakcanary/f684db0b-69c5-4179-971d-aba25ee73596_pending.hprof, createdAtTimeMillis=1562732527492, analysisDurationMillis=5884, retainedInstances=[LeakingInstance(referenceKey=170097c6-0c7d-4d13-b205-731808b0378f, referenceName=, instanceClassName=androidx.constraintlayout.widget.ConstraintLayout, watchDurationMillis=-5242926, exclusionStatus=null, leakTrace=
    ┬
    ├─ android.view.inputmethod.InputMethodManager
    │    Leaking: NO (it's a GC root and a class is never leaking)
    │    ↓ static InputMethodManager.sInstance
    ├─ android.view.inputmethod.InputMethodManager
    │    Leaking: NO (RecyclerView↓ is not leaking)
    │    ↓ InputMethodManager.mNextServedView
    ├─ androidx.recyclerview.widget.RecyclerView
    │    Leaking: NO (ContributorAdapter↓ is not leaking and View attached)
    │    View#mAttachInfo is not null (view attached)
    │    View#mParent is set
    │    View.mID=R.id.contributor_list (2131230806)
    │    View.mWindowAttachCount=1
    │    ↓ RecyclerView.mAdapter
    ├─ com.android.example.github.ui.repo.ContributorAdapter
    │    Leaking: NO (RepoFragment$onViewCreated$adapter$1↓ is not leaking)
    │    ↓ ContributorAdapter.callback
    ├─ com.android.example.github.ui.repo.RepoFragment$onViewCreated$adapter$1
    │    Leaking: NO (RepoFragment↓ is not leaking)
    │    Anonymous subclass of kotlin.jvm.internal.Lambda
    │    ↓ RepoFragment$onViewCreated$adapter$1.this$0
    ├─ com.android.example.github.ui.repo.RepoFragment
    │    Leaking: NO (FragmentManagerImpl↓ is not leaking and RepoFragment#mFragmentManager is not null)
    │    ↓ RepoFragment.mFragmentManager
    ├─ androidx.fragment.app.FragmentManagerImpl
    │    Leaking: NO (HashMap↓ is not leaking)
    │    ↓ FragmentManagerImpl.mActive
    ├─ java.util.HashMap
    │    Leaking: NO (HashMap$Node[]↓ is not leaking)
    │    ↓ HashMap.table
    ├─ java.util.HashMap$Node[]
    │    Leaking: NO (HashMap$Node↓ is not leaking)
    │    ↓ array HashMap$Node[].[12]
    ├─ java.util.HashMap$Node
    │    Leaking: NO (SearchFragment↓ is not leaking)
    │    ↓ HashMap$Node.value
    ├─ com.android.example.github.ui.search.SearchFragment
    │    Leaking: NO (SearchFragment#mFragmentManager is not null)
    │    ↓ SearchFragment.binding$delegate
    │                     ~~~~~~~~~~~~~~~~
    ├─ com.android.example.github.util.AutoClearedValue
    │    Leaking: UNKNOWN
    │    ↓ AutoClearedValue._value
    │                       ~~~~~~
    ├─ com.android.example.github.databinding.SearchFragmentBindingImpl
    │    Leaking: UNKNOWN
    │    ↓ SearchFragmentBindingImpl.mRoot
    │                                ~~~~~
    ╰→ androidx.constraintlayout.widget.ConstraintLayout
    ​     Leaking: YES (RefWatcher was watching this)
    , retainedHeapSize=null)])
```

It could be because the AutoClearedValue is not working as it was intended. This may be related to the comments in this issue: https://github.com/googlesamples/android-architecture-components/issues/69

Also, is it considered a bad practice to declare a Fragment's dataBinding variable as a member variable? Do we always need to wrap then in something like AutoClearedValue?

**Comment by finneapps** [comment]
I guess this case happens when you put your Fragment to the backstack and only onDestryView() is called and not onDestroy()?

**Comment by timatoe** [comment]
@finneapps Yep, I think that's what's happening here. What I want to know is, should AutoClearedValue be modified so that it clears the values onDestroyView() instead of onDestroy()?

**Comment by ianhanniballake** [comment]
Yep, this was fixed in #785
